File: GenesisAndroid/src/test/java/com/emerchantpay/gateway/genesisandroid/internal/ReconcileRequestUnitTest.java
package com.emerchantpay.gateway.genesisandroid.internal;

import android.content.Context;
import android.test.mock.MockContext;

import com.emerchantpay.gateway.genesisandroid.api.internal.request.ReconcileRequest;
import com.emerchantpay.gateway.genesisandroid.api.models.PaymentAddress;

import org.junit.Before;
import org.junit.Test;

import java.net.MalformedURLException;
import java.util.ArrayList;
import java.util.UUID;

import static org.junit.Assert.assertEquals;

public class ReconcileRequestUnitTest {

    private Context context;

    private ReconcileRequest request;
    private PaymentAddress address;

    private ArrayList<String> transactionTypes;

    @Before
    public void mockParams() throws IllegalAccessException, MalformedURLException {
        // Application context
        context = new MockContext().getApplicationContext();

        // Generate unique Id
        String uniqueId = UUID.randomUUID().toString();

        // Payment request
        request = new ReconcileRequest();
        request.setUniqueId(uniqueId);
    }

    @Test
    public void testReconcile() throws IllegalAccessException {
        assertEquals(request.getTransactionType(), "wpf_reconcile");
    }

    @Test
    public void testBuildRequest() {
        assertEquals(request.toXML(), request.getRequest().toXML());
    }
}

File: GenesisAndroid/src/test/java/com/emerchantpay/gateway/genesisandroid/internal/ReconcileRequestUnitTest.kt
package com.emerchantpay.gateway.genesisandroid.internal

import android.content.Context
import com.emerchantpay.gateway.genesisandroid.api.internal.request.ReconcileRequest
import com.emerchantpay.gateway.genesisandroid.api.models.PaymentAddress
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import java.net.MalformedURLException
import java.util.*

class ReconcileRequestUnitTest {

    @Mock
    private val context: Context? = null

    private var request: ReconcileRequest? = null
    private val address: PaymentAddress? = null

    private val transactionTypes: ArrayList<String>? = null

    @Before
    @Throws(IllegalAccessException::class, MalformedURLException::class)
    fun mockParams() {
        // Generate unique Id
        val uniqueId = UUID.randomUUID().toString()

        // Payment request
        request = ReconcileRequest()
        request!!.setUniqueId(uniqueId)
    }

    @Test
    @Throws(IllegalAccessException::class)
    fun testReconcile() {
        assertEquals(request!!.getTransactionType(), "wpf_reconcile")
    }

    @Test
    fun testBuildRequest() {
        assertEquals(request!!.toXML(), request!!.request.toXML())
    }
}
